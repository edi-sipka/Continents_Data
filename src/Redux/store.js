/* eslint-disable */

import {
  configureStore,
  applyMiddleware,
  combineReducers,
} from '@reduxjs/toolkit';
import countryReducers from './reducers';
import thunk from 'redux-thunk';
import categorySelector from './category';

const rootReducer = combineReducers({
  countries: countryReducers,
  category: categorySelector,
});

const store = configureStore({ reducer: rootReducer }, applyMiddleware(thunk));
export default store;
