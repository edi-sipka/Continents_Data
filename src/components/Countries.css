.row {
  list-style: none;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}
.lists {
  background-color: rgba(35, 1, 36, 0.43);
  box-shadow: 5px 10px #aeaeae4e;
  display: flex;
  flex-direction: column;
}

.lists:hover {
  transform: scale(1.1);
  transition: all 0.5s ease-in;
}
.image-position {
  width: 300%;
  display: flex;
  align-items: center;
  justify-content: center;
}
.item {
  display: flex;
  gap: 20px;
  margin-top: 10px;
  margin-left: 20px;
}
.icons {
  font-size: 24px;
  margin-left: 20px;
}
