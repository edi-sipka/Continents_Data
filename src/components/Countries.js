import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';

import { BsFillArrowRightCircleFill } from 'react-icons/bs';
import { aboutCountries } from '../Redux/reducers';
import './Countries.css';

const Countries = () => {
  const countries = useSelector((state) => state.countries);
  const dispatch = useDispatch();
  const handleClick = (name) => {
    dispatch(aboutCountries(name));
  };

  return (
    <div className="countries">
      <ul className="row">
        {' '}
        <>
          {countries.map((country) => (
            <li key={country?.name.common} className="lists">
              <div className="list">
                <div className="icons">
                  <h2>{country?.name.common}</h2>
                  <Link to={`/:/${country?.name.common}`}>
                    <BsFillArrowRightCircleFill
                      className="btn"
                      onClick={() => handleClick(country?.name.common)}
                    />
                  </Link>
                  <div className="image-position">
                    <img src={country?.flags.png} alt="flag" />
                  </div>
                </div>
              </div>

              <div className="item">
                <p>Capital:</p>
                <p>{country?.capital}</p>
              </div>
              <div className="item">
                <p>SubRegion:</p>
                <p>{country?.subregion}</p>
              </div>
              <div className="item">
                <p>TimeZone:</p>
                <p>{country?.timezones[0]}</p>
              </div>
            </li>
          ))}
        </>
      </ul>
    </div>
  );
};

export default Countries;
