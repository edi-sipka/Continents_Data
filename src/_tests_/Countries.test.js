/* eslint-disable */

import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import Countries from '../components/Countries';

describe('Navbar rendering', () => {
  test('Navbar rendering', () => {
    const render = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Countries />
        </BrowserRouter>
      </Provider>
    );
    expect(render).toMatchSnapshot();
  });
});
