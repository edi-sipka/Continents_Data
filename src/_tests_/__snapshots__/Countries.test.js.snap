// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar rendering Navbar rendering 1`] = `
<div
  className="countries"
>
  <ul
    className="row"
  >
     
  </ul>
</div>
`;
